Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -325,88 +325,9 @@ public class RepositoryPackageContainer: PackageContainer, CustomStringConvertib
     }
 }
 
-/// Adaptor for exposing repositories as PackageContainerProvider instances.
-///
-/// This is the root class for bridging the manifest & SCM systems into the
-/// interfaces used by the `DependencyResolver` algorithm.
-public class RepositoryPackageContainerProvider: PackageContainerProvider {
```

Comment:
@tomerd @abertelrud After digging into the implementation a bit more, I hit a fork in the road and was interested to hear your ideas for how best to proceed.

Currently, responsibility for creating package containers falls to `RepositoryPackageContainerProvider`, which conforms to the `PackageContainerProvider` protocol. 

My first instinct was to either create a complement to `RepositoryPackageContainerProvider` for registry packages, or to update that class to handle both repository and registry packages. However, I noticed that `RepositoryPackageContainerProvider` doesn't do much beyond copying some state from a `Workspace` and implementing the `getContainer` method requirement.

So my thinking here was that we could remove a layer of indirection by making `Workspace` conform to `PackageContainerProvider` directly. This way, we can access all of the information we need directly, (like registry configuration, netrc credentials, etc.).

Do you think this is the right approach to take?